packages/permission-controller/src/PermissionController.ts

```diff
@@ -1889,8 +1908,32 @@ export class PermissionController<
       permissions: requestedPermissions,
     };
 
+    const approvedRequest = await this.requestUserApproval(permissionsRequest);
     const { permissions: approvedPermissions, ...requestData } =
-      await this.requestUserApproval(permissionsRequest);
+      approvedRequest;
+
+    const sideEffects = this.getSideEffects(approvedPermissions);
+
+    if (sideEffects.permittedHandlers.length > 0) {
+      const sideEffectsData = await this.executeSideEffects(
+        sideEffects,
+        approvedRequest,
+      );
+      const mappedData = Object.keys(approvedPermissions).reduce(
+        (acc, permission, i) => ({ [permission]: sideEffectsData[i], ...acc }),
```

Comment:
It seems like the `executeSideEffects` function returns an array of side effect results, for all the provided onPermitted handlers. But here, it looks like we are mapping those results against each permission by index, even if that permission didn't have any side effects.

So as an example, if you requested 10 permissions, and only the last 5 had side effects, would the first five without side effects be assigned the side effect data, and the remaining 5 with side effects get undefined?

Reply:
Good catch !